### PR TITLE
#117667 add a markup escape for filenames in error messages

### LINKED ISSUE #117667: Text rendered with markup in the file rename error modal box
- VS Code Version: Insiders February 2021 (version 1.54)
- OS Version: Arco Linux v21.01.7

Steps to Reproduce:

1. try to rename a file with an invalid name
2. it renders markup in the modal box
image
image

Does this issue occur when all extensions are disabled?: Yes

### PR DESCRIPTION
**Pull request by cloud11665**
This PR fixes #117667
then:
image
now:
image

### CODE REVIEW

File: src/vs/workbench/contrib/files/browser/fileActions.ts [hunk 1]
```diff
@@ -710,6 +711,16 @@ export function validateFileName(item: ExplorerItem, name: string): { content: s
 	return null;
 }
 
+function escapeMarkdown(filename: string): string {
+	if (!filename) {
```
Comment (isidorn) [inline src/vs/workbench/contrib/files/browser/fileActions.ts:715]: No need for this check we are using non nullable types

**Review by isidorn — CHANGES_REQUESTED** [review]

**Comment by isidorn** [comment]
Thanks for your PR, but do we have to use regex for this?
Is it possible to just check where this content is being used and there is probably a flag to not render markdown but regular text.
That would be a simpler fix

File: src/vs/workbench/contrib/files/browser/fileActions.ts (hunk 1 above)
Comment (Cloud11665) [reply src/vs/workbench/contrib/files/browser/fileActions.ts:715]: without the check, it didnt pass [CI](https://github.com/microsoft/vscode/runs/1979231440)

**Comment by Cloud11665** [comment]
> Is it possible to just check where this content is being used and there is probably a flag to not render markdown but regular text.

This wouldnt work, because markdown is used [here](https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/files/browser/fileActions.ts#L688) to make the line bold.

**Comment by Cloud11665** [comment]
@isidorn

**Comment by pibuxd** [comment]
same issue

**Comment by isidorn** [comment]
@Cloud11665 thanks for the PR, but I do not find the solution elegant. Not sure if there is a better way - maybe not. But for now I would not merge this. Maybe next milestone, thus assigning to backlog.
Thank you again for working on this

**Comment by isidorn** [comment]
No plans to merge, thus closing. Thank you for the PR either way and sorry for not being able to merge this in.

**Pull request closed without merging**